src/main/java/vn/zalopay/benchmark/GRPCSampler.java

```diff
@@ -70,37 +73,53 @@ private void initGrpcClient() {
 
     @Override
     public SampleResult sample(Entry ignored) {
-        GrpcResponse grpcResponse = new GrpcResponse();
         SampleResult sampleResult = new SampleResult();
 
-        // Console prints unknown exceptions - Intercepts unknown exceptions and sends them to the console to print instead of throwing the results into the GRPC response.
+        // Intercepts exceptions before GRPC requests are initiated
         try {
+            sampleResult.setSampleLabel(getName());
             initGrpcConfigRequest();
             initGrpcClient();
-            sampleResult.setSampleLabel(getName());
             String grpcRequest = clientCaller.buildRequestAndMetadata(getRequestJson(), getMetadata());
             sampleResult.setSamplerData(grpcRequest);
             sampleResult.setRequestHeaders(clientCaller.getMetadataString());
             sampleResult.sampleStart();
         } catch (Exception e) {
-            log.error("An unknown error occurred before the GRPC request was initiated, and the stack trace is as follows: ", e);
-            startSamplerToStopIfCatchException(sampleResult);
-            generateErrorResultInInitGRPCRequest(sampleResult, e);
+            sampleResult.setSuccessful(false);
+            sampleResult.setResponseCode(" 400");
+            sampleResult.setDataType(SampleResult.TEXT);
+            sampleResult.setResponseMessage(" GRPCSampler parsing exception: An unknown exception occurred before the GRPC request was initiated, See response body for the stack trace.");
+            sampleResult.setResponseData(ExceptionUtils.getPrintExceptionToStr(e, null), "UTF-8");
             return sampleResult;
         }
 
         // Initiate a GRPC request
-        try {
-            grpcResponse = clientCaller.call(getDeadline());
-            sampleResult.sampleEnd();
+        GrpcResponse grpcResponse = clientCaller.call(getDeadline());
+        sampleResult.sampleEnd();
+        sampleResult.setDataType(SampleResult.TEXT);
+        if (grpcResponse.isSuccess()) {
             sampleResult.setSuccessful(true);
             sampleResult.setResponseCodeOK();
-            sampleResult.setResponseMessage("Success");
-            sampleResult.setDataType(SampleResult.TEXT);
+            sampleResult.setResponseMessage(" success");
             sampleResult.setResponseData(grpcResponse.getGrpcMessageString().getBytes(StandardCharsets.UTF_8));
-        } catch (RuntimeException e) {
-            startSamplerToStopIfCatchException(sampleResult);
-            generateErrorResult(grpcResponse, sampleResult, e);
+        } else {
```

Comment:
It would be better if we extract this method to a smaller method function such as grpcRequestProcessing

Reply:
Hello, this exception handling code is temporarily not reusable, so directly written inside the method, easier to understand more intuitive